### ISSUE TITLE
KotlinGenerator produces invalid syntax for PostgreSQL enum types

### ISSUE THREAD

**Issue by dsvensson** [jOOQ/jOOQ#10761]
### Expected behavior

Generated code:

```kotlin
override fun getCatalog(): Catalog? = getSchema()?.getCatalog()
```

### Actual behavior

Generated code:

```kotlin
override fun getCatalog(): Catalog? = getSchema() == null ? null : getSchema().getCatalog()
```

### Steps to reproduce the problem

Add an enum in a PostgreSQL database and generate Kotlin source code.

### Versions

- jOOQ: 3.14
- Java: 13
- Kotlin: 1.4.10
- Database (include vendor): PostgreSQL 11.6
- OS: Linux
- JDBC Driver (include name if inofficial driver): org.postgresql:postgresql:42.2.12
- Gradle plugin: etiennestuder/gradle-jooq-plugin v5.2

**Comment by lukaseder** [comment]
Thanks a lot for your report. Our integration tests only covered enums without schema (H2, MySQL), which is why this was undetected. Will fix ASAP for jOOQ 3.15.0 and 3.14.1 (#10765).

**Comment by dsvensson** [comment]
Right... when following the code, it's apparent that it's not nullable so it should just use `getSchema().getCatalog()` right away.

**Comment by lukaseder** [comment]
This slipped by 3.14.1 - sorry for the inconvenience. Will fix this now

**Comment by lukaseder** [comment]
> Right... when following the code, it's apparent that it's not nullable so it should just use `getSchema().getCatalog()` right away.

Agreed

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0 and 3.14.2 (#10765).

**Comment by jippeholwerda** [comment]
> Fixed for jOOQ 3.15.0 and 3.14.2 (#10765).

Cool! When do you expect 3.14.2 will be released?

**Comment by lukaseder** [comment]
> > Fixed for jOOQ 3.15.0 and 3.14.2 (#10765).
> 
> Cool! When do you expect 3.14.2 will be released?

Yes
